src/parser/core/modules/PrecastAction.ts

```diff
@@ -0,0 +1,38 @@
+import {Event, isAbilityEvent, isCastEvent, isDamageEvent} from 'fflogs'
+import Module from 'parser/core/Module'
+
+// Actions that initiate a pull are only a damage event - no cast
+// Fabricate fake cast events to clean up the mess
+export default class PrecastAction extends Module {
+	static handle = 'precastAction'
+	static debug = false
+
+	normalise(events: Event[]): Event[] {
+		const startTime = this.parser.fight.start_time
+
+		for (const event of events) {
+			this.debug(`Timestamp: ${event.timestamp} - Event Type: ${String(event.type)} - Action: ${isAbilityEvent(event) ? event.ability.name : ''}`)
+
+			// Only care about events by the player and ignore autoattacks
+			if (!this.parser.byPlayer(event) || !isAbilityEvent(event) || event.ability.name === 'attack')
```

Comment:
This'll fail for ranged phys (i think?) - their auto is `shot`.

If you dep on `data` (should be more safe to do so), and `data.getAction`, should be able to check `action.autoAttack` which is `boolean | undefined`